platform/linux-generic/odp_crypto.c

```diff
@@ -146,6 +171,8 @@ odp_crypto_generic_session_t *alloc_session(void)
 	}
 	odp_spinlock_unlock(&global->lock);
 
+	session->idx = (session - global->sessions) / sizeof(*session);
+
```

Comment:
The cipher_valid and hmac_valid bits are not cleared at session allocation, so they can be left to what they were in an old and freed session that occupied the same memory slot.

Reply:
Whole `odp_crypto_generic_session_t` data is cleared during `odp_crypto_session_destroy()`

Reply:
ok, I can see it now. I think it would be less fragile to clear the session at alloc() instead of relying it to be cleared at init_global() and destroy().

Reply:
I did not say the keys should not be cleared at destroy() but that it would be better (as in, more difficult to introduce a bug later) if the alloc did not rely on the clearing of the valid bits etc having happened earlier (in destroy or in global init). But this is not a sticking point for me.

Btw, if you are concerned about storing keys in memory for too long, then this patch does just that by keeping the openssl contexts alive until local_term().

Reply:
Ok, I will add some cleaning to _create stage. At lease it is not a fast path operation, so performance should not matter that much.